tests/baselines/reference/privateNamesNotAllowedAsParameters.errors.txt

```diff
@@ -0,0 +1,21 @@
+tests/cases/conformance/classes/members/privateNames/privateNamesNotAllowedAsParameters.ts(4,12): error TS1138: Parameter declaration expected.
+tests/cases/conformance/classes/members/privateNames/privateNamesNotAllowedAsParameters.ts(4,24): error TS1005: ';' expected.
+tests/cases/conformance/classes/members/privateNames/privateNamesNotAllowedAsParameters.ts(4,26): error TS1068: Unexpected token. A constructor, method, accessor, or property was expected.
+tests/cases/conformance/classes/members/privateNames/privateNamesNotAllowedAsParameters.ts(5,1): error TS1128: Declaration or statement expected.
+
+
+==== tests/cases/conformance/classes/members/privateNames/privateNamesNotAllowedAsParameters.ts (4 errors) ====
+    // @target es6
+    
+    class A {
+        setFoo(#foo: string) {}
+               ~~~~
+!!! error TS1138: Parameter declaration expected.

```

Comment:
like the variable declaration, this should be accepted in the parser and rejected in the grammar-checking functions of the checker.

Reply:
In https://github.com/microsoft/TypeScript/pull/30829#issuecomment-509839268 we decided to save improving this kind of error message for a follow-on. The architecture will take some work, since it's hard to do with the approach we've chosen, but I agree it's important